composeApp/src/commonMain/kotlin/com/arkivanov/minesweeper/game/GameStore.kt

```diff
@@ -211,3 +213,11 @@ private fun GameState.toggleFlagIntent(location: Location): GameState {
 
     return copy(grid = grid + (location to cell.copy(status = status.copy(isFlagged = !status.isFlagged))))
 }
+
+private fun GameState.tick(): GameState {
+    if (timer >= 999) {
+        return this
+    }
+
+    return copy(timer = timer + 1)
```

Comment:
Could be more Kotlin-idiomatic `if (timer < 999) copy(timer = timer + 1) else this`, and you can also use `=` instead of `return`.